packages/x-data-grid/src/utils/css/context.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useGridConfiguration } from '../../hooks/utils/useGridConfiguration';
+
+const CLASSNAME_PREFIX = 'MuiDataGridVariables';
+
+const CSSVariablesContext = React.createContext({
+  className: 'unset',
+  tag: <style href="/unset" />,
+});
+
+export function useCSSVariablesClass() {
+  return React.useContext(CSSVariablesContext).className;
+}
+
+export function useCSSVariables() {
+  return React.useContext(CSSVariablesContext);
+}
+
+export function GridPortalWrapper({ children }: { children: React.ReactNode }) {
+  const className = useCSSVariablesClass();
+  return <div className={className}>{children}</div>;
+}
+
+export function GridCSSVariablesContext(props: { children: any }) {
+  const config = useGridConfiguration();
+  const description = config.hooks.useCSSVariables();
+
+  const context = React.useMemo(() => {
+    const className = `${CLASSNAME_PREFIX}-${description.id}`;
+    const cssString = `.${className}{${variablesToString(description.variables)}}`;
+    const tag = <style href={`/${className}`}>{cssString}</style>;
+    return { className, tag };
```

Comment:
Is this SSR-compatible?

Reply:
Yes, inline `<style>` tags are SSR compatible. The `description.id` is a stable hash so there's also no hydration error.